arcgis-ios-sdk-samples/Content Display Logic/Controllers/OptionsTableViewController.swift

```diff
@@ -31,30 +31,66 @@ class OptionsTableViewController: UITableViewController {
         self.options = options
         self.selectedIndex = selectedIndex
+        self.allowsEmptySelection = allowsEmptySelection
         self.onChange = onChange
         super.init(nibName: nil, bundle: nil)
     }
     
-    /// Creates a new instance with the given labels, selected index, and
+    /// Creates a new instance with the options, selected index, empty selection bool, and selection
+    /// change handler.
+    /// - Parameters:
+    ///   - options: The options displayed by the view controller.
+    ///   - selectedIndex: The index of the currently selected option or `nil`.
+    ///   - onChange: A closure called when the selected option has changed.
+    convenience init(options: [Option], selectedIndex: Int, onChange: @escaping (Int) -> Void) {
+        self.init(options: options, selectedIndex: selectedIndex, allowsEmptySelection: false, onChange: { onChange($0!) })
+    }
+    
+    /// Creates a new instance with the given labels, selected index, empty selection bool, and
+    /// selection change handler.
+    /// - Parameters:
+    ///   - labels: An array of labels for the options.
+    ///   - selectedIndex: The index of the currently selected option or `nil`.
+    ///   - onChange: A closure called when the selected option has changed.
+    convenience init(labels: [String], selectedIndex: Int, onChange: @escaping (Int) -> Void) {
+        let options = labels.map { Option(label: $0) }
+        self.init(options: options, selectedIndex: selectedIndex, allowsEmptySelection: false, onChange: { onChange($0!) })
+    }
+    
+    /// Creates a new instance with the given labels, selected index, empty selection bool, and
     /// selection change handler.
     /// - Parameters:
     ///   - labels: An array of labels for the options.
     ///   - selectedIndex: The index of the currently selected option or `nil`.
     ///   - onChange: A closure called when the selected option has changed.
     convenience init(labels: [String], selectedIndex: Int?, onChange: @escaping (Int) -> Void) {
         let options = labels.map { Option(label: $0) }
-        self.init(options: options, selectedIndex: selectedIndex, onChange: onChange)
+        self.init(options: options, selectedIndex: selectedIndex, allowsEmptySelection: false, onChange: { onChange($0!) })
+    }
+    
+    /// Creates a new instance with the given labels, selected index, empty selection bool, and
+    /// selection change handler.
+    /// - Parameters:
+    ///   - labels: An array of labels for the options.
+    ///   - selectedIndex: The index of the currently selected option or `nil`.
+    ///   - allowsEmptySelection: A bool that determines whether or not empty selection is allowed.
+    ///   - onChange: A closure called when the selected option has changed.
+    convenience init(labels: [String], selectedIndex: Int?, allowsEmptySelection: Bool, onChange: @escaping (Int?) -> Void) {
```

Comment:
I decided to include the `allowsEmptySelection` parameter here to easily differentiate this init from the others. 

There is one other case (non optional `selectedIndex` with optional `onChange`) that I didn't include because I don't think it would be useful but if anyone disagrees, I can add that as well.

Reply:
This initializer isn't needed. Instead, the code in `SaveAsViewController.swift` should be updated to account for these changes:

![image](https://user-images.githubusercontent.com/2257493/127205878-759f448d-6317-46bb-9f12-3dc90ed5990b.png)

Reply:
@philium By removing this initializer, how will the `onChange` return an optional value? And wouldn't that defeat the purpose of adding the `allowsEmptySelection` property? The other initializers set `allowsEmptySelection` to false and `onChange` cannot return nil

Reply:
Oops! Not this one. I meant the one before it, where the optionality of the `selectedIndex` parameter and the index taken by the `onChange` closure do not match:

```swift
convenience init(labels: [String], selectedIndex: Int?, onChange: @escaping (Int) -> Void)
```

Reply:
Oh I added this one because of situations like[ this](https://github.com/Esri/arcgis-runtime-samples-ios/blob/ee1bbe9b20a009770c12a3f2fb048edf552115e9/arcgis-ios-sdk-samples/Utility%20network/Configure%20subnetwork%20trace/ConfigureSubnetworkTraceOptionsViewController.swift#L166-L178). There were a handful of similar cases where `selectedIndex` can be nil due to no initial selection but would later need to return a non optional index.

Reply:
It still think some of these initializers can be consolidated, but I don't have time to look into it more at the moment.